src/mozanalysis/experiment.py

```diff
@@ -654,37 +696,67 @@ def _build_exposure_query(
                 raise ValueError(
                     "App ID must be defined for building Cirrus exposures query"
                 )
+            if not self.experimental_unit == ExperimentalUnit.CLIENT:
+                raise IncompatibleExperimentalUnit(
+                    "Cirrus exposures currently only support client_id analysis units"
+                )
             return self._build_exposure_query_glean_event(
                 time_limits,
                 self.app_id,
                 client_id_field='mozfun.map.get_key(event.extra, "user_id")',
                 event_category="cirrus_events",
             )
         else:
-            raise ValueError
+            assert_never(exposure_query_type)
 
     def _build_enrollments_query_normandy(
-        self, time_limits: TimeLimits, sample_size: int = 100
+        self,
+        time_limits: TimeLimits,
+        sample_size: int = 100,
     ) -> str:
         """Return SQL to query enrollments for a normandy experiment"""
-        return f"""
-        SELECT
-            e.client_id,
-            `mozfun.map.get_key`(e.event_map_values, 'branch')
-                AS branch,
-            MIN(e.submission_date) AS enrollment_date,
-            COUNT(e.submission_date) AS num_enrollment_events
-        FROM
-            `moz-fx-data-shared-prod.telemetry.events` e
-        WHERE
-            e.event_category = 'normandy'
-            AND e.event_method = 'enroll'
-            AND e.submission_date
-                BETWEEN '{time_limits.first_enrollment_date}' AND '{time_limits.last_enrollment_date}'
-            AND e.event_string_value = '{self.experiment_slug}'
-            AND e.sample_id < {sample_size}
-        GROUP BY e.client_id, branch
-            """  # noqa:E501
+        if self.experimental_unit == ExperimentalUnit.CLIENT:
+            return f"""
+            SELECT
+                e.client_id,
+                `mozfun.map.get_key`(e.event_map_values, 'branch')
+                    AS branch,
+                MIN(e.submission_date) AS enrollment_date,
+                COUNT(e.submission_date) AS num_enrollment_events
+            FROM
+                `moz-fx-data-shared-prod.telemetry.events` e
+            WHERE
+                e.event_category = 'normandy'
+                AND e.event_method = 'enroll'
+                AND e.submission_date
+                    BETWEEN '{time_limits.first_enrollment_date}' AND '{time_limits.last_enrollment_date}'
+                AND e.event_string_value = '{self.experiment_slug}'
+                AND e.sample_id < {sample_size}
```

Comment:
I believe sample_id is currently based on client_id, do we know if that will continue into the future? Will there be a group_sample_id or something that will be based on the profile_group_id or group_id column?

Reply:
Ohh yikes, good question... group_ids are not guaranteed to all fall into the same sample_ids... that's a huge footgun

Reply:
Confirming this is an open question with a Jira ticking coming to track the work around it and ongoing discussions are happening among DS and DE.

Reply:
Filed [a ticket](https://mozilla-hub.atlassian.net/jira/software/c/projects/DS/boards/1947?selectedIssue=DS-3857) to follow up on this, not going to block this PR on it.